### ISSUE TITLE
Citation autocomplete documentation

### ISSUE THREAD

**Issue by TryerGit** [lervag/vimtex#2513]
The documentation at: https://github.com/lervag/vimtex/blob/master/doc/vimtex.txt

states: 

`Citation completion is triggered by \cite{ commands.`

My thought on this was that once the user types in insert mode (or via a snippet) `\cite{` a list of all possible citations would pop up. However, this does not seem to be the case. This list of all possible citations pops up only after the user presses the following `\cite{<CTRL-X><CTRL-O>`

This also seems to be implied by the examples provided in the documentation:

        \cite{Knuth 1981<CTRL-X><CTRL-O>
        \cite{algo<CTRL-X><CTRL-O>
        \cite{Don.*Knuth<CTRL-X><CTRL-O>

I wanted to confirm that indeed the expected behaviour is to have the triggering of the popup with suggestions happens only after pressing of `<CTRL-X><CTRL-O>` by the user and not just bare `\cite{`.

Thank you.

**Comment by lervag** [comment]
Yes, that is correct. This is at least partially explained just above at `:help vimtex-completion`. You may also be interested in reading `:help vimtex-complete-auto`.

**Comment by lervag** [comment]
I've updated the docs slightly, I hope that helps. See `:help vimtex-completion`.

**Comment by TryerGit** [comment]
Thanks @lervag That is good to know. I just wanted to ensure that I was not having some screwed up setting that does not exhibit intended behaviour.

I use the default YCM settings for id-based completion of previously existing words in my buffered .tex file. This happens by default with nothing in way of settings in `.vimrc`. I then noticed that you suggest the following in the documentation:

```
if !exists('g:ycm_semantic_triggers')
    let g:ycm_semantic_triggers = {}
  endif
  au VimEnter * let g:ycm_semantic_triggers.tex=g:vimtex#re#youcompleteme
```

I turned this on, but was unable to find any benefit. For e.g., typing `\begin{enumerate}` does not give rise to an automatic `\end{enumerate}` and so on. So, I am wondering what the intended benefit is of the above suggested settings for ycm and vimtex in the `.vimrc`?

Ideally, with YCM and Vimtex, especially as it pertains to Bibtex citation autocompletion, it would be great if `\cite{` autotriggers the popup of bibkeys instead of having to Type CtrlX Ctrl O. I don't know if this would be possible and if it would be whether it would be a request to place on this site or over at the YCM site.
